### PR TITLE
tests: POST /login returns same device_id

### PR DESCRIPTION
**Pull request by meenal06**
Signed-off-by: Meenal Trivedi <meenaltrivedi6102@gmail.com>

### CODE REVIEW

**Comment by meenal06** [comment]
Hi @Kegsay @callahad , Please review the following tests. I have also created a new method to create dummy users to use them for login tests.

File: tests/apidoc_login_test.go [hunk 1]
```diff
@@ -41,34 +42,76 @@ func TestLogin(t *testing.T) {
...
 					match.JSONKeyTypeEqual("access_token", gjson.String),
-					match.JSONKeyEqual("home_server", "hs1"),
+					match.JSONKeyEqual("device_id", deviceId),
 				},
 			})
 		})
 	})
 }
+
+func createDummyUser(t *testing.T, unauthedClient *client.CSAPI, userId string) error {
```
Comment (Kegsay) [inline tests/apidoc_login_test.go:93]: When specifying acronyms, please use upper case letters e.g `userID` not `userId`. This is what many Go developers expect to see, and some linters will fail code without this.

File: tests/apidoc_login_test.go [hunk 2]
```diff
@@ -41,34 +42,76 @@ func TestLogin(t *testing.T) {
...
+	reqBody, err := json.Marshal(map[string]interface{}{
+		"auth": map[string]string{
+			"type": "m.login.dummy",
+		},
+		"username": userId,
+		"password": "superuser",
+	})
+	if err != nil {
+		t.Log("unable to marshal json")
+		return err
```
Comment (Kegsay) [inline tests/apidoc_login_test.go:103]: When making helper functions like this one, it helps to do as much as possible. Look at all the call sites for `createDummyUser` and see how they all then do `if err != nil` followed by failing the test. I would just change `createDummyUser` to not return `error` and instead do the check here and fail the test if there is an error. This reduces boilerplate required in the test.

**Review by Kegsay — APPROVED** [review]
Minor things, but otherwise looks good to me (LGTM)!

File: tests/apidoc_login_test.go (hunk 2 above)
Comment (meenal06) [reply tests/apidoc_login_test.go:103]: Thanks for the tip, I will be making corrections right away!

**Comment by meenal06** [comment]
Hi @Kegsay , I have the changes you requested please review them

File: tests/apidoc_login_test.go [hunk 3]
```diff
@@ -41,34 +42,67 @@ func TestLogin(t *testing.T) {
...
-					match.JSONKeyTypeEqual("user_id", gjson.String),
+					match.JSONKeyEqual("home_server", "hs1"),
 				},
 			})
-			res = unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "login"}, json.RawMessage(`{
+		})
+		// sytest: POST /login returns the same device_id as that in the request
+		t.Run("POST /login returns the same device_id as that in the request", func(t *testing.T) {
+			t.Parallel()
+			deviceId := "test_device_id"
```
Comment (Kegsay) [inline tests/apidoc_login_test.go:65]: Again, please use `deviceID`.

**Review by Kegsay — APPROVED** [review]

File: tests/apidoc_login_test.go (hunk 3 above)
Comment (meenal06) [reply tests/apidoc_login_test.go:65]: Apologies, let me fix it quickly 😓

**Comment by meenal06** [comment]
Fixed the deviceID @Kegsay , please review

**Review by Kegsay — APPROVED** [review]
LGTM!

**Pull request merged**